Fix an issue with CUnit where failed tests are counted once for each failed assertion.

diff -r 08cdddd4e18c bitten/build/ctools.py
--- a/bitten/build/ctools.py	Fri Apr 10 15:00:11 2009 +0200
+++ b/bitten/build/ctools.py	Fri Apr 10 15:05:37 2009 +0200
@@ -245,11 +245,20 @@
                             continue
                         suiteName = suite.children ('SUITE_NAME').next().gettext()
                         info ("%s [%s]" % ("*" * (57 - len (suiteName)), suiteName))
+                        testResults = set()
                         for record in suite.children ('CUNIT_RUN_TEST_RECORD'):
                             for result in record.children():
                                 if result.name not in ('CUNIT_RUN_TEST_SUCCESS', 'CUNIT_RUN_TEST_FAILURE'):
                                     continue
                                 testName = result.children ('TEST_NAME').next().gettext()
+                                if testName in testResults:
+                                    if result.name == 'CUNIT_RUN_TEST_FAILURE':
+                                        error ("%s(%d): %s"
+                                                   % (result.children ('FILE_NAME').next().gettext(),
+                                                      int (result.children ('LINE_NUMBER').next().gettext()),
+                                                      result.children ('CONDITION').next().gettext()))
+                                    continue
+                                testResults.add (testName)
                                 info ("Running %s..." % testName);
                                 test = xmlio.Element('test')
                                 test.attr['fixture'] = suiteName
